speechbrain/dataio/dataio.py

```diff
@@ -28,6 +31,50 @@
 logger = logging.getLogger(__name__)
 
 
+def load_audio_pyav(uri, num_frames=-1, frame_offset=0, normalize=True):
+    """Mimics torchaudio.load using PyAV which bundles ffmpeg.
+
+    Arguments
+    ---------
+    uri: str
+        Path to file
+    num_frames: int
+        Number of frames to load. Default (-1) loads all frames
+    frame_offset: int
+        Frame from which to start loading. Default (0) loads all frames
+    normalize: bool
+        Whether to normalize output to float32 if int32
+    """
+
+    raw_buffer = io.BytesIO()
+    dtype = None
+    sample_rate = None
+    channels = None
+
+    with av.open(uri, mode="r", metadata_errors="ignore") as container:
+        for frame in container.decode(audio=0):
+            sample_rate = frame.rate
+            layout = frame.layout
+            channels = len(layout.channels)
+            array = frame.to_ndarray()
+            dtype = array.dtype
+
+            raw_buffer.write(array)
+
+    audio = np.frombuffer(raw_buffer.getbuffer(), dtype=dtype)
+    audio = np.reshape(audio, [channels, -1], order="F")
+
+    if dtype.kind == "i" and normalize:
+        audio = audio.astype(np.float32) / np.iinfo(dtype).max
+
+    if num_frames >= 0:
```

Comment:
Not sure how much of a proof-of-concept this currently is but I'll leave some comments anyway.

Would it be possible to implement `frame_offset`/`num_frames` without needing to decode the entire file? Certain datasets I have encountered have long files with utterances representing a very small segment of the file.

Reply:
I did think about this, and its probably important to have it, but I'm not sure how straightforward it is to do it for compressed audio files.